source/shared/cpp/ObjectModel/CollectionTypeElement.cpp

```diff
@@ -0,0 +1,159 @@
+    {
+        SetPadding(!(context.GetParentalContainerStyle() == ContainerStyle::None &&
+            GetStyle() == ContainerStyle::Default));
+    } 
+    else 
+    {
+        SetPadding(false);
+    }
+}
+
+bool CollectionTypeElement::GetBleed() const
+{
+    return m_hasBleed;
+}
+
+void CollectionTypeElement::SetBleed(const bool value)
+{
+    m_hasBleed = value;
+}
+
+// Applies Bleed Flag When appropriate
+void CollectionTypeElement::ConfigBleed(const AdaptiveCards::ParseContext& context)
+{
+    // we allows bleed when self has padding and at least one parent has padding
+    const AdaptiveSharedNamespace::InternalId id = context.PaddingParentInternalId();
+    const bool canBleed = GetBleed() && (id != AdaptiveSharedNamespace::InternalId::Invalid);
+    if (canBleed)
+    {
+        SetParentalId(id);
+    }
+    SetCanBleed(canBleed);
+}
+
+void CollectionTypeElement::SetCanBleed(const bool value)
+{
+    m_canBleed = value;
+}
+
+bool CollectionTypeElement::GetCanBleed() const
+{
+    return m_canBleed;
+}
+
+void CollectionTypeElement::SetParentalId(const AdaptiveSharedNamespace::InternalId &id)
+{
+    m_parentalId = id;
+}
+
+AdaptiveSharedNamespace::InternalId CollectionTypeElement::GetParentalId(void) const
+{
+    return m_parentalId;
+}
+
+// convinience method for configuring for container style
+void CollectionTypeElement::ConfigForContainerStyle(const ParseContext& context)
+{
+    ConfigPadding(context);
+    ConfigBleed(context);
+}
+
+std::shared_ptr<BackgroundImage> CollectionTypeElement::GetBackgroundImage() const
+{
+    return m_backgroundImage;
+}
+
+void CollectionTypeElement::SetBackgroundImage(const std::shared_ptr<BackgroundImage> value)
+{
+    m_backgroundImage = value;
+}
+
+std::shared_ptr<BaseActionElement> CollectionTypeElement::GetSelectAction() const
+{
+    return m_selectAction;
+}
+
+void CollectionTypeElement::SetSelectAction(const std::shared_ptr<BaseActionElement> action)
+{
+    m_selectAction = action;
+}
+
+Json::Value CollectionTypeElement::SerializeToJsonValue() const
+{
+    Json::Value root = BaseCardElement::SerializeToJsonValue();
+
+    if (m_selectAction != nullptr)
+    {
+        root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::SelectAction)] =
+            BaseCardElement::SerializeSelectAction(m_selectAction);
+    }
+
+    if (m_backgroundImage != nullptr && !m_backgroundImage->GetUrl().empty())
+    {
+        root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::BackgroundImage)] = m_backgroundImage->SerializeToJsonValue();
+    }
+
+    if (GetStyle() != ContainerStyle::None)
+    {
+        root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Style)] = ContainerStyleToString(GetStyle());
+    }
+
+    if (GetBleed())
+    {
+        root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Bleed)] = true;
+    }
+
+    return root;
+}
+
+void CollectionTypeElement::DeserializeChildren(ParseContext&, const Json::Value&)
+{
+    return;
+}
```

Comment:
do we really want to provide a default implementation here? perhaps it should be pure virtual...